api.go

```diff
@@ -33,21 +33,26 @@ type APIErrorMessage struct {
 	Error  string `json:"error"`
 }
 
-func createError(errorMsg string) []byte {
-	errorObj := APIErrorMessage{"error", errorMsg}
-	responseMsg, err := json.Marshal(&errorObj)
-
-	if err != nil {
-		log.Error("Couldn't marshal error stats: ", err)
-	}
+func apiOk(msg string) APIErrorMessage {
+	return APIErrorMessage{"ok", msg}
+}
 
-	return responseMsg
+func apiError(msg string) APIErrorMessage {
+	return APIErrorMessage{"error", msg}
 }
 
-func doJSONWrite(w http.ResponseWriter, code int, responseMessage []byte) {
+func doJSONWrite(w http.ResponseWriter, code int, obj interface{}) {
+	// TODO(mvdan): remove a couple of weeks after the api refactor,
```

Comment:
Without the comment, or without the comment and panic?

I can't ensure it easily right now. I guess I could statically, but I would need to write up a tool to do static analysis and check all the possible types for the interface parameter. I've been playing with `guru` for a while now, but I can't get it to work properly - it seems to think the possible types are three, so it's clearly wrong.

A far easier option is to have this panic for a little bit and see if any of us run into a panic, which I would quickly fix. Is that a problem?